ql/math/matrix.hpp

```diff
@@ -60,6 +60,9 @@ namespace QuantLib {
 
         Matrix& operator=(const Matrix&);
         Matrix& operator=(Matrix&&) noexcept;
+
+        bool operator==(const Matrix&) const;
+        bool operator!=(const Matrix&) const;
```

Comment:
For consistency with other binary operators like `+` and `*`, I'd write these as free functions instead of methods.

Reply:
Happy to change it. I saw that the equality operators for the Array are methods, while the other binary operators are free functions. shall we change them too?